sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveExternalCatalog.scala

```diff
@@ -320,6 +322,7 @@ private[spark] class HiveExternalCatalog(conf: SparkConf, hadoopConf: Configurat
       }
 
       table.copy(
+        schema = getHiveCompatibleSchema(table.schema),
```

Comment:
Can we add some comments saying that we convert the table schema at the last step here, so that the table schema JSON string in the table properties still uses the original schema.

Reply:
actually i liked you previous comment about converting the schema at the similar place so I refactored table property creation to be done before the call to `createDataSourceTable`